### PR TITLE
Add table azure_mssql_virtual_machine. Closes #275

### LINKED ISSUE #275: Add table azure_mssql_virtual_machine
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_virtual_machine

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro azure-test % ./tint.js azure_mssql_virtual_machine
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_mssql_virtual_machine []

PRETEST: tests/azure_mssql_virtual_machine

TEST: tests/azure_mssql_virtual_machine
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_mssql_virtual_machine.named_test_resource will be created
  + resource "azurerm_mssql_virtual_machine" "named_test_resource" {
      + id                               = (known after apply)
      + r_services_enabled               = true
      + sql_connectivity_port            = 1433
      + sql_connectivity_type            = "PRIVATE"
      + sql_connectivity_update_password = (sensitive value)
      + sql_connectivity_update_username = (sensitive value)
      + sql_license_type                 = "PAYG"
      + virtual_machine_id               = (known after apply)

      + auto_patching {
          + day_of_week                            = "Sunday"
          + maintenance_window_duration_in_minutes = 60
          + maintenance_window_starting_hour       = 2
        }
    }

  # azurerm_network_interface.named_test_resource will be created
  + resource "azurerm_network_interface" "named_test_resource" {
      + applied_dns_servers           = (known after apply)
      + dns_servers                   = (known after apply)
      + enable_accelerated_networking = false
      + enable_ip_forwarding          = false
      + id                            = (known after apply)
      + internal_dns_name_label       = (known after apply)
      + internal_domain_name_suffix   = (known after apply)
      + location                      = "westeurope"
      + mac_address                   = (known after apply)
      + name                          = "turbottest63414"
      + private_ip_address            = (known after apply)
      + private_ip_addresses          = (known after apply)
      + resource_group_name           = "turbottest63414"
      + virtual_machine_id            = (known after apply)

      + ip_configuration {
          + name                          = "internal"
          + primary                       = (known after apply)
          + private_ip_address            = (known after apply)
          + private_ip_address_allocation = "dynamic"
          + private_ip_address_version    = "IPv4"
          + subnet_id                     = (known after apply)
        }
    }

  # azurerm_resource_group.named_test_resource will be created
  + resource "azurerm_resource_group" "named_test_resource" {
      + id       = (known after apply)
      + location = "westeurope"
      + name     = "turbottest63414"
    }

  # azurerm_subnet.named_test_resource will be created
  + resource "azurerm_subnet" "named_test_resource" {
      + address_prefix                                 = (known after apply)
      + address_prefixes                               = [
          + "10.0.2.0/24",
        ]
      + enforce_private_link_endpoint_network_policies = false
      + enforce_private_link_service_network_policies  = false
      + id                                             = (known after apply)
      + name                                           = "turbottest63414"
      + resource_group_name                            = "turbottest63414"
      + virtual_network_name                           = "turbottest63414"
    }

  # azurerm_virtual_machine.named_test_resource will be created
  + resource "azurerm_virtual_machine" "named_test_resource" {
      + availability_set_id              = (known after apply)
      + delete_data_disks_on_termination = false
      + delete_os_disk_on_termination    = false
      + id                               = (known after apply)
      + license_type                     = (known after apply)
      + location                         = "westeurope"
      + name                             = "turbottest63414"
      + network_interface_ids            = (known after apply)
      + resource_group_name              = "turbottest63414"
      + vm_size                          = "Standard_DS1_v2"

      + identity {
          + identity_ids = (known after apply)
          + principal_id = (known after apply)
          + type         = (known after apply)
        }

      + os_profile {
          + admin_password = (sensitive value)
          + admin_username = "exampleadmin"
          + computer_name  = "winhost01"
          + custom_data    = (known after apply)
        }

      + os_profile_windows_config {
          + enable_automatic_upgrades = true
          + provision_vm_agent        = true
          + timezone                  = "Pacific Standard Time"
        }

      + storage_data_disk {
          + caching                   = (known after apply)
          + create_option             = (known after apply)
          + disk_size_gb              = (known after apply)
          + lun                       = (known after apply)
          + managed_disk_id           = (known after apply)
          + managed_disk_type         = (known after apply)
          + name                      = (known after apply)
          + vhd_uri                   = (known after apply)
          + write_accelerator_enabled = (known after apply)
        }

      + storage_image_reference {
          + offer     = "SQL2017-WS2016"
          + publisher = "MicrosoftSQLServer"
          + sku       = "SQLDEV"
          + version   = "latest"
        }

      + storage_os_disk {
          + caching                   = "ReadOnly"
          + create_option             = "FromImage"
          + disk_size_gb              = (known after apply)
          + managed_disk_id           = (known after apply)
          + managed_disk_type         = "Premium_LRS"
          + name                      = "turbottest63414"
          + os_type                   = (known after apply)
          + write_accelerator_enabled = false
        }
    }

  # azurerm_virtual_network.named_test_resource will be created
  + resource "azurerm_virtual_network" "named_test_resource" {
      + address_space         = [
          + "10.0.0.0/16",
        ]
      + dns_servers           = (known after apply)
      + guid                  = (known after apply)
      + id                    = (known after apply)
      + location              = "westeurope"
      + name                  = "turbottest63414"
      + resource_group_name   = "turbottest63414"
      + subnet                = (known after apply)
      + vm_protection_enabled = false
    }

Plan: 6 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + resource_id      = (known after apply)
  + resource_name    = "turbottest63414"
  + sql_license_type = "PAYG"
  + subscription_id  = "d46d7416-f95f-4771-bbb5-529d4c76659c"
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 2s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414]
azurerm_virtual_network.named_test_resource: Creating...
azurerm_virtual_network.named_test_resource: Creation complete after 10s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.Network/virtualNetworks/turbottest63414]
azurerm_subnet.named_test_resource: Creating...
azurerm_subnet.named_test_resource: Creation complete after 5s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.Network/virtualNetworks/turbottest63414/subnets/turbottest63414]
azurerm_network_interface.named_test_resource: Creating...
azurerm_network_interface.named_test_resource: Creation complete after 5s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.Network/networkInterfaces/turbottest63414]
azurerm_virtual_machine.named_test_resource: Creating...
azurerm_virtual_machine.named_test_resource: Still creating... [10s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [20s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [30s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [40s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [50s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [1m0s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [1m10s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [1m20s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [1m30s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [1m40s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [1m50s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [2m0s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [2m10s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [2m20s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [2m30s elapsed]
azurerm_virtual_machine.named_test_resource: Still creating... [2m40s elapsed]
azurerm_virtual_machine.named_test_resource: Creation complete after 2m41s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.Compute/virtualMachines/turbottest63414]
azurerm_mssql_virtual_machine.named_test_resource: Creating...
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [10s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [20s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [30s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [40s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [50s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [1m0s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [1m10s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [1m20s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [1m30s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [1m40s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [1m50s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [2m0s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [2m10s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [2m20s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [2m30s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [2m40s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [2m50s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [3m0s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [3m10s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [3m20s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [3m30s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [3m40s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [3m50s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [4m0s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [4m10s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [4m20s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [4m30s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [4m40s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [4m50s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [5m0s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [5m10s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Still creating... [5m20s elapsed]
azurerm_mssql_virtual_machine.named_test_resource: Creation complete after 5m25s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines/turbottest63414]

Apply complete! Resources: 6 added, 0 changed, 0 destroyed.

Outputs:

resource_id = "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines/turbottest63414"
resource_name = "turbottest63414"
sql_license_type = "PAYG"
subscription_id = "d46d7416-f95f-4771-bbb5-529d4c76659c"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines/turbottest63414",
    "name": "turbottest63414",
    "resource_group": "turbottest63414",
    "sql_server_license_type": "PAYG",
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "type": "Microsoft.SqlVirtualMachine/sqlVirtualMachines"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest63414/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines/turbottest63414",
    "name": "turbottest63414",
    "resource_group": "turbottest63414",
    "sql_server_license_type": "PAYG",
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "type": "Microsoft.SqlVirtualMachine/sqlVirtualMachines"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "name": "turbottest63414",
    "title": "turbottest63414"
  }
]
✔ PASSED

POSTTEST: tests/azure_mssql_virtual_machine

TEARDOWN: tests/azure_mssql_virtual_machine

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
arnab@turbotindias-MacBook-Pro steampipe-plugin-azure % steampipe query
Welcome to Steampipe v0.7.3
For more information, type .help
> select
  id,
  name,
  type,
  provisioning_state,
  sql_image_offer,
  sql_server_license_type,
  region
from
  azure_mssql_virtual_machine;
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+---------------------
| id                                                                                                                                                                  | name          | type                
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+---------------------
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines/test-mssql-vm | test-mssql-vm | Microsoft.SqlVirtual
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+---------------------
> select
  id,
  name,
  type,
  provisioning_state
from
  azure_mssql_virtual_machine
where
  provisioning_state = 'Failed';
+----+------+------+--------------------+
| id | name | type | provisioning_state |
+----+------+------+--------------------+
+----+------+------+--------------------+
> 
```
</details>

### CODE REVIEW

File: docs/tables/azure_mssql_virtual_machine.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  type,
+  provisioning_state,
+  sql_image_offer,
+  sql_server_license_type,
+  region
+from
+  azure_mssql_virtual_machine;
+```
+
+### List failed load balancers
```
Comment (bigdatasourav) [inline docs/tables/azure_mssql_virtual_machine.md:22]: Update the example

File: azure/table_azure_mssql_virtual_machine.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	if err != nil {
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+
+	client := sqlvirtualmachine.NewSQLVirtualMachinesClient(subscriptionID)
+	client.Authorizer = session.Authorizer
+
+	result, err := client.List(ctx)
+	if err != nil {
```
Comment (ParthaI) [inline azure/table_azure_mssql_virtual_machine.go:184]: Add error logger

File: azure/table_azure_mssql_virtual_machine.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+
+	client := sqlvirtualmachine.NewSQLVirtualMachinesClient(subscriptionID)
+	client.Authorizer = session.Authorizer
+
+	op, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
```
Comment (ParthaI) [inline azure/table_azure_mssql_virtual_machine.go:228]: Add error logger

File: azure/table_azure_mssql_virtual_machine.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+		return nil, err
+	}
+	
+	for _, virtualMachine := range result.Values() {
+		d.StreamListItem(ctx, virtualMachine)
+	}
+
+	for result.NotDone() {
+		err = result.NextWithContext(ctx)
+		if err != nil {
```
Comment (ParthaI) [inline azure/table_azure_mssql_virtual_machine.go:194]: add error logger

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**